crates/cli/src/opts/build/core.rs

```diff
@@ -135,6 +135,38 @@ impl CoreBuildArgs {
     pub fn get_remappings(&self) -> Vec<Remapping> {
         self.project_paths.get_remappings()
     }
+
+    /// Merge the current cli arguments into the specified zksync configuration
+    fn apply_zksync_overrides(&self, mut zksync: ZkSyncConfig) -> ZkSyncConfig {
+        let compiler_cfg = &self.compiler;
+
+        macro_rules! set_if_some {
+            ($src:expr, $dst:expr) => {
+                if let Some(src) = $src {
+                    $dst = src.into();
+                }
+            };
+        }
+
+        set_if_some!(compiler_cfg.zksync.then_some(true), zksync.enable);
```

Comment:
If we are keeping the cli maybe prefix everything with `zksync` and the same name as the config.

Reply:
We could also move zksync specific stuff to another struct and do the figment magic there instead?

Reply:
yeah and `#[serde(flatten)]` it too, nice

Reply:
Nice, maybe also rename `zksync` toml item to `zk` just so everything is named the same.

Reply:
I would have put `zksync` in the cli also, but some keys are already long enough, but in the config it makes it very easy to see this is _zksync_ specific configuration